core/src/main/java/com/nisovin/magicspells/spells/command/ScrollSpell.java

```diff
@@ -228,19 +238,30 @@ public ItemStack createScroll(Spell spell, int uses, ItemStack item) {
 
 		return item;
 	}
-	
+
 	@Override
-	public List<String> tabComplete(CommandSender sender, String[] args) {
-		if (sender instanceof ConsoleCommandSender) {
-			if (args.length == 1) return TxtUtil.tabCompletePlayerName(sender);
-			if (args.length == 2) return TxtUtil.tabCompleteSpellName(sender);
-			if (args.length == 3) return List.of("1");
-		}
-		else if (sender instanceof Player) {
-			if (args.length == 1) return TxtUtil.tabCompleteSpellName(sender);
-			if (args.length == 2) return List.of("1");
-		}
-		return null;
+	public @NotNull Iterable<@NotNull String> stringSuggestions(@NotNull CommandContext<CommandSourceStack> context, @NotNull CommandInput input) {
+		CommandSourceStack stack = context.sender();
+
+		CommandSender executor = Objects.requireNonNullElse(stack.getExecutor(), stack.getSender());
+		if (executor instanceof Player caster)
+			return OwnedSpellParser.suggest(caster, requireTeachPerm ? MagicSpells.getSpellbook(caster)::canTeach : null);
+
+		if (!(executor instanceof ConsoleCommandSender)) return Collections.emptyList();
+
+		List<String> suggestions = TxtUtil.tabCompletePlayerName(executor);
+		CommandInput original = input.copy();
+
+		String playerName = input.readString();
+		if (playerName.isEmpty() || input.isEmpty() && !input.input().endsWith(" ")) return suggestions;
+
+		Player player = Bukkit.getPlayer(playerName);
+		if (player == null) return suggestions;
+
+		String diff = original.difference(input.skipWhitespace(), true);
+		SpellParser.suggest().stream().map(spell -> diff + spell).forEach(suggestions::add);
```

Comment:
When a player is found, `suggestions` shouldn't suggest player names anymore for the next parameter, just spells.